easybuild/easyblocks/f/flexiblas.py

```diff
@@ -0,0 +1,166 @@
+        return extra_vars
+
+    def __init__(self, *args, **kwargs):
+        """Easyblock constructor."""
+        super(EB_FlexiBLAS, self).__init__(*args, **kwargs)
+
+        build_dep_names = set(dep['name'] for dep in self.cfg.dependencies(build_only=True))
+        dep_names = [dep['name'] for dep in self.cfg.dependencies()]
+        self.blas_libs = [x for x in dep_names if x not in build_dep_names]
+
+    def configure_step(self):
+        """Custom configuration for FlexiBLAS, based on which BLAS libraries are included as dependencies."""
+
+        configopts = {
+            'BLAS_AUTO_DETECT': ('OFF', 'ON')[bool(self.cfg['blas_auto_detect'])],
+            'LAPACK': ('OFF', 'ON')[bool(self.cfg['enable_lapack'])],
+            'FLEXIBLAS_DEFAULT': self.cfg['flexiblas_default'] or self.blas_libs[0],
+        }
+
+        supported_blas_libs = ['BLIS', 'NETLIB', 'OpenBLAS']
+
+        # make sure that default backend is a supported library
+        flexiblas_default = configopts['FLEXIBLAS_DEFAULT']
+        if flexiblas_default not in supported_blas_libs:
+            raise EasyBuildError("%s not in list of supported BLAS libs %s", flexiblas_default, supported_blas_libs)
+
+        # make sure that all backends are supported libraries
+        unsupported_libs = [x for x in self.blas_libs if x not in supported_blas_libs]
+        if unsupported_libs:
+            raise EasyBuildError("One or more unsupported libraries used: %s", ', '.join(unsupported_libs))
+
+        # list of BLAS libraries to use is specified via -DEXTRA=...
+        configopts['EXTRA'] = ';'.join(self.blas_libs)
+
+        # for each library, we have to specify how to link to it via -DXXX_LIBRARY;
+        # see https://github.com/mpimd-csc/flexiblas#setup-with-custom-blas-and-lapack-implementations
+        for blas_lib in self.blas_libs:
+            key = '%s_LIBRARY' % blas_lib
+            configopts[key] = blas_lib.lower()
+
+        # only add configure options to configopts easyconfig parameter if they're not defined yet,
+        # to allow easyconfig to override specific settings
+        for key, value in sorted(configopts.items()):
+            opt = '-D%s=' % key
+            if key not in self.cfg['configopts']:
+                self.cfg.update('configopts', opt + "'%s'" % value)
+
+        builddir = os.path.join(self.builddir, 'easybuild_obj')
+
+        # specify compiler commands with absolute paths, to ensure that RPATH wrapper scripts are used
+        if build_option('rpath'):
+            self.cfg['abs_path_compilers'] = True
+
+            # inject RPATH link option for build location of libflexiblas.so.3,
+            # to ensure that test binaries can find the FlexiBLAS library
+            build_libdir = os.path.join(builddir, 'lib')
+            for envvar in ['CFLAGS', 'CXXFLAGS', 'FFLAGS']:
+                flags = os.getenv(envvar)
+                setvar(envvar, '-Wl,-rpath=%s %s' % (build_libdir, flags))
```

Comment:
This leaves the buildpath inside the RPATH (although admittedly only as the last option). Instead of setting this, can't you set
```
pretestopts = "LD_LIBRARY_PATH=%s:$LD_LIBRARY_PATH " % os.path.join(self.builddir, 'easybuild_obj', 'lib')
```
for the tests below?

Reply:
Well, no, because `$LD_LIBRARY_PATH` is irrelevant when RPATH is used...

I see your point though, ideally this is only used when building the test, so I can maybe do this via `pretestopts`, I'll look into that...